src/v2/Apps/ArtistSeries/Components/ArtistSeriesArtworksFilter.tsx

```diff
@@ -0,0 +1,158 @@
+import { ArtistSeriesArtworksFilter_artistSeries } from "v2/__generated__/ArtistSeriesArtworksFilter_artistSeries.graphql"
+import { BaseArtworkFilter } from "v2/Components/v2/ArtworkFilter"
+import { ArtworkFilterContextProvider } from "v2/Components/v2/ArtworkFilter/ArtworkFilterContext"
+import { updateUrl } from "v2/Components/v2/ArtworkFilter/Utils/urlBuilder"
+import { Match, RouterState, withRouter } from "found"
+import React from "react"
+import { RelayRefetchProp, createRefetchContainer, graphql } from "react-relay"
+
+interface ArtistSeriesArtworksFilterProps {
+  artistSeries: ArtistSeriesArtworksFilter_artistSeries
+  relay: RelayRefetchProp
+  match?: Match
+}
+
+const ArtistSeriesArtworksFilter: React.FC<ArtistSeriesArtworksFilterProps> = props => {
+  const { match, relay, artistSeries } = props
+  const { filtered_artworks } = artistSeries
+
+  const hasFilter = filtered_artworks && filtered_artworks.id
+
+  // If there was an error fetching the filter,
+  // we still want to render the rest of the page.
+  if (!hasFilter) return null
+
+  return (
+    <ArtworkFilterContextProvider
+      filters={match && match.location.query}
+      sortOptions={[
+        { value: "-decayed_merch", text: "Default" },
+        { value: "-has_price,-prices", text: "Price (desc.)" },
+        { value: "-has_price,prices", text: "Price (asc.)" },
+        { value: "-partner_updated_at", text: "Recently updated" },
+        { value: "-published_at", text: "Recently added" },
+        { value: "-year", text: "Artwork year (desc.)" },
+        { value: "year", text: "Artwork year (asc.)" },
+      ]}
+      onChange={updateUrl}
+    >
+      <BaseArtworkFilter
+        relay={relay}
+        viewer={artistSeries}
+        relayVariables={{
+          aggregations: ["TOTAL"],
+        }}
+      ></BaseArtworkFilter>
+    </ArtworkFilterContextProvider>
```

Comment:
Nice, so this wasn't so bad I hope. The v2 artwork filter requires a lot of relay setup unfortunately, but once the compiler passes it will ususally just work.

Out of curiosity, did we try to use the main `ArtworkFilter` component (which wraps the context provider)? [See here.](https://github.com/artsy/force/blob/master/src/v2/Apps/Collect/Routes/Collect/index.tsx#L91) All of the component parts were exported so that we could get around different kinds of query requirements but this is generally all that's needed.

Reply:
Yeah not too bad! It mostly just worked like you said once we got it going. 

We needed to be able to pass an artist series slug to the filter query which `ArtworkFilter` doesn't support, so we went with using `BaseArtworkFilter` instead.